### PR TITLE
fix: escape properly column names on GBQ [TCTC-6519]

### PR DESCRIPTION
**Pull request by PrettyWood**
## BEFORE

Screenshot 2023-08-07 at 22 51 04

## AFTER

Test passes ✅

### CODE REVIEW

**Review by Sanix-Darker — APPROVED** [review]

File: server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py [hunk 1]
```diff
@@ -54,23 +63,24 @@ class GoogleBigQueryQueryBuilder(QueryBuilder):
...
+            dialect=SQLDialect.GOOGLEBIGQUERY, wrapper_cls=GoogleBigQueryValueWrapper, **kwargs
+        )
+
+    @staticmethod
+    def wrap_constant(val, wrapper_cls=GoogleBigQueryValueWrapper):
+        return super().wrap_constant(val, wrapper_cls)
+
+    @staticmethod
+    def wrap_json(val, wrapper_cls=GoogleBigQueryValueWrapper):
+        return super().wrap_json(val, wrapper_cls)
```
Comment (Sanix-Darker) [inline server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py:76]: a small UT to represent these new changes?

Comment (PrettyWood) [reply server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py:76]: It's still a WIP yes don't worry!

File: server/src/weaverbird/backends/pypika_translator/translators/base.py [hunk 2]
```diff
@@ -252,6 +252,12 @@ def _merge_first_steps(
 
         return ctx
 
+    def _ensure_term_uses_wrapper(self: Self, term: Term):
```
Comment (PrettyWood) [inline server/src/weaverbird/backends/pypika_translator/translators/base.py:255]: This is the real fix ! This is ugly but unfortunately there is no easy way to fix this
(see https://github.com/kayak/pypika/issues/622)

File: server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrettyWood) [reply server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py]: Nevermind I commented it. Better to have a hotfix fast and we'll dig after

File: server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py [hunk 3]
```diff
@@ -54,23 +63,24 @@ class GoogleBigQueryQueryBuilder(QueryBuilder):
     QUERY_CLS = GoogleBigQueryQuery
 
     def __init__(self, **kwargs: Any) -> None:
-        super().__init__(dialect=SQLDialect.GOOGLEBIGQUERY, **kwargs)
+        super().__init__(
+            dialect=SQLDialect.GOOGLEBIGQUERY, wrapper_cls=GoogleBigQueryValueWrapper, **kwargs
+        )
+
+    # @staticmethod
```
Comment (WilliamGorge) [inline server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py:70]: Code commente a cleaner?

Comment (PrettyWood) [reply server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py:70]: je l'ai laissé commenté pour revenir dessus dans un 2nd temps et ne pas oublier

**Review by fspot — APPROVED** [review]

File: server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py (hunk 3 above)
Comment (PrettyWood) [reply server/src/weaverbird/backends/pypika_translator/translators/googlebigquery.py]: Nevermind c'est useless pour nous

**Pull request merged**